mesheryctl/internal/cli/root/workspaces/create.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2024 Layer5, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package workspaces
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/config"
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/system"
+	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
+	"github.com/layer5io/meshery/server/models"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var CreateWorkspaceCmd = &cobra.Command{
+	Use:   "create",
+	Short: "Create a new workspaces",
+	Long:  `Create a new workspaces by providing the name and description of the workspace`,
+	Example: `
+// Create a new workspace
+mesheryctl exp workspace create orgId --name [workspace-name] --description [workspace-description]
+
+// Documentation for workspace can be found at:
+https://docs.layer5.io/cloud/spaces/workspaces/
+`,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		//Check prerequisite
+
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return utils.ErrLoadConfig(err)
+		}
+		err = utils.IsServerRunning(mctlCfg.GetBaseMesheryURL())
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+		ctx, err := mctlCfg.GetCurrentContext()
+		if err != nil {
+			utils.Log.Error(system.ErrGetCurrentContext(err))
+			return err
+		}
+		err = ctx.ValidateVersion()
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+		return nil
+	},
+
+	Args: func(cmd *cobra.Command, args []string) error {
+		const errMsg = "Usage: mesheryctl exp workspace create --orgId [orgId] --name [workspace-name] --description [workspace-description]\nRun 'mesheryctl exp workspace create --help' to see detailed help message"
+
+		// Check if all three flags are set
+		orgIdFlag, _ := cmd.Flags().GetString("orgId")
+		nameFlag, _ := cmd.Flags().GetString("name")
+		descriptionFlag, _ := cmd.Flags().GetString("description")
+
+		if orgIdFlag == "" || nameFlag == "" || descriptionFlag == "" {
+			return errors.New(utils.WorkspaceSubError(fmt.Sprintf("'%s' is an invalid subcommand. %s\n", args[0], errMsg), "create"))
```

Comment:
Can't we just use cmd.Usage? 
Also this isn't meshkit error.

Reply:
Yes, we can use cmd.Usage(). That will be more convenient to user.
Thanks for suggestion.